### PR TITLE
Demo

### PR DESCRIPTION
**Pull request by yehtarnsu**
Demo

### CODE REVIEW

File: lambda-assets/device-activator/app.ts [hunk 1]
```diff
@@ -94,11 +94,9 @@ export const handler = async (event: any = {}) : Promise <any> => {
       }),
     );
 
-    let payloadString: string = '';
-    payload.forEach(num => {
-      payloadString += String.fromCharCode(num);
-    });
-    const { body } = JSON.parse(payloadString);
+    const { body } = JSON.parse(
+      [...payload].map((num: number) => String.fromCharCode(num)).join(''),
```
Comment (MinCheTsai) [inline lambda-assets/device-activator/app.ts:98]: Can change to String.fromCharCode.apply(null, [...payload])?

File: lambda-assets/ca-registrator/app.ts [hunk 2]
```diff
@@ -84,7 +84,7 @@ export const handler = async (event: any = {}) : Promise <any> => {
       organizationUnitName: '',
     };
 
-    const verifiers: {[key: string]: string} = JSON.parse(process.env.VERIFIERS!)
+    const verifiers: {[key: string]: string} = [...JSON.parse(process.env.VERIFIERS!)]
```
Comment (MinCheTsai) [inline lambda-assets/ca-registrator/app.ts:87]: left is object, right value is array?

File: src/demo/demo1.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import * as apigateway from '@aws-cdk/aws-apigateway';
```
Comment (MinCheTsai) [inline src/demo/demo1.ts:1]: demo1.ts change to demo-for-what?

demo-jitr.ts?

File: src/demo/demo2.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+const app = new cdk.App();
+const anotherStack = new cdk.Stack(app, 'anotherStack');
+const myVault = new s3.Bucket(anotherStack, 'myVault');
+
+const id = 'JitrDemo2';
+const stack = new cdk.Stack(app, id);
+const jitr = new JustInTimeRegistration(stack, id, {
+  vault: {
+    bucket: myVault,
+    prefix: 'my/ca/path',
```
Comment (MinCheTsai) [inline src/demo/demo2.ts:20]: 合併做一次展示完畢即可

File: src/demo/demo3.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * The device activator will ask the CA-specified verifier
+ * if the device is suspicious or not, and make the decision
+ * on the device activation.
+ */
+
+const app = new cdk.App();
+const anotherStack = new cdk.Stack(app, 'anotherStack');
+
+const myVerifier = new lambda.Function(anotherStack, 'verifier1', {
```
Comment (MinCheTsai) [inline src/demo/demo3.ts:19]: 合併做一次展示完畢即可

File: README.md [hunk 6]
```diff
@@ -1,13 +1,56 @@
...
+    npm install @softchef/cdk-iot-security
+
+### Yarn
+
+    yarn add @softchef/cdk-iot-security
+    
+
+## Just-in-Time Registration
+
+![](https://drive.google.com/uc?export=view&id=1pq8N6uDgTxA9qxgeBPptPJvDFE_hc0lM)
```
Comment (MinCheTsai) [inline README.md:18]: image can commit to git, path is /docs(new folder)

File: src/demo/demo1.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+const id = 'JitrDemo1';
+const stack = new cdk.Stack(app, id);
+const jitr = new JustInTimeRegistration(stack, id, {
+  vault: {
+    bucket: myVault,
+  },
+});
+const api = new apigateway.RestApi(stack, id);
+api.root.addResource('caRegister').addMethod('POST', new apigateway.LambdaIntegration(jitr.caRegistrator));
+api.root.addResource('verifiersFetch').addMethod('GET', new apigateway.LambdaIntegration(jitr.verifiersFetcher));
```
Comment (MinCheTsai) [inline src/demo/demo1.ts:28]: How to use this demo?

1. Deploy demo(ignore basic skill)
2. Register CA certificate(by CURL)
3. Git clone IoT demo client(NodeJs/EmbedC) 這個再討論
4. Generate device certificate in IoT demo client
5. Connect to AWS IoT, checking the rule, activator, verifier are worked.
6. Done. The device already join to IoT solution :)

File: lambda-assets/ca-registrator/app.ts (hunk 2 above)
Comment (yehtarnsu) [reply lambda-assets/ca-registrator/app.ts:87]: 換行有reduce

File: src/demo/demo1.ts (hunk 7 above)
Comment (yehtarnsu) [reply src/demo/demo1.ts]: 1.

**Review by MinCheTsai — APPROVED** [review]

**Pull request closed without merging**